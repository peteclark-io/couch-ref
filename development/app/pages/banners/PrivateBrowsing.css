@import "../../css/branding.css";

.splash {
   position: absolute;
   top: 0;
   left: 0;
   width: 100%;
   height: 100%;
   background-color: var(--brand-danger);
   text-align: center;
}

.container {
   margin: 0 4vw;
}

.error {
   font-family: var(--font-primary);
   color: var(--brand-secondary);
   font-size: 6em;
   margin-top: 6vh;
}

.message {
   font-family: var(--font-secondary);
   color: var(--brand-secondary);
   margin-top: 40px;
   font-size: 1.6rem;
}

.message a {
   color: white;
   text-decoration: underline;
}

.emoji {
  font-size: 0.5em;
}

.tldr {
   font-family: var(--font-secondary);
   color: var(--brand-secondary);
   margin-top: 60px;
   font-size: 2.8rem;
}
