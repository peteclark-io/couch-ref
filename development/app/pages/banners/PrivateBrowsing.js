'use strict';

import React from 'react';
import styles from './PrivateBrowsing.css';

const PrivateBrowsing = React.createClass({

   render: function() {
      return (
         <div className={styles.splash}>
            <h1 className={styles.error}>Warning</h1>
            <h3 className={styles.message}>We've detected that you're using a private browser, most likely via an iOS device.</h3>
            <h3 className={styles.message}>Unfortunately, we're not currently able to provide the full CouchRef experience from within a private browser, as it prevents us from using some important browser functionality (namely <a href="https://www.html5rocks.com/en/features/storage">Local Storage</a>).</h3>
            <h3 className={styles.tldr}>If you'd like to use CouchRef (and we really hope you do), please switch to a regular browser window.</h3>
         </div>
      );
   }
});

export default PrivateBrowsing;
