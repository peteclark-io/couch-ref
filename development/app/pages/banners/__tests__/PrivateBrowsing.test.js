'use strict';

import React from 'react';
import PrivateBrowsing from '../PrivateBrowsing';
import renderer from 'react-test-renderer';

test('PrivateBrowsing.', () => {
  const component = renderer.create(
    <PrivateBrowsing />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
