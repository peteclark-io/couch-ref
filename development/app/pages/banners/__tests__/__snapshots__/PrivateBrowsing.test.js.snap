exports[`test PrivateBrowsing. 1`] = `
<div
  className="splash">
  <div
    className="container">
    <h1
      className="error">
      Warning
    </h1>
    <h3
      className="message">
      We\'ve detected that you\'re using a private browser, most likely via an iOS device.
    </h3>
    <h3
      className="message">
      Unfortunately, we\'re not currently able to provide the full CouchRef experience from within a private browser, as it prevents us from using some important browser functionality (namely 
      <a
        href="https://www.html5rocks.com/en/features/storage">
        Local Storage
      </a>
      ).
    </h3>
    <h3
      className="tldr">
      If you\'d like to use CouchRef (and we really hope you do), please switch to a regular browser window.
    </h3>
  </div>
</div>
`;
